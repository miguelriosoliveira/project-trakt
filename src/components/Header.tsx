import Image from 'next/image';
import Link from 'next/link';
import { FiSearch, FiMenu } from 'react-icons/fi';

export function Header() {
	return (
		<header className="flex items-center justify-between fixed w-full p-2 bg-black bg-opacity-60 transition-all duration-700 z-10 hover:bg-opacity-90">
			<FiSearch color="white" size={20} strokeWidth={3} />

			<Link href="/">
				<Image
					alt="Trakt logo: a red circle with an inclined letter T"
					src="/assets/logo.webp"
					width={40}
					height={40}
				/>
			</Link>

			<div className="flex gap-2 items-center">
				<FiMenu color="white" size={25} strokeWidth={3} />
				<Image
					alt="The user's profile picture"
					src="https://github.com/miguelriosoliveira.png"
					width={35}
					height={35}
					className="rounded-full"
				/>
			</div>
		</header>
	);
}
