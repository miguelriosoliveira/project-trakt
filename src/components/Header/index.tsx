import Image from 'next/image';
import Link from 'next/link';
import { FiSearch, FiMenu } from 'react-icons/fi';

import styles from './styles.module.scss';

export function Header() {
	return (
		<header className={styles.container}>
			<FiSearch color="white" size={20} strokeWidth={3} />

			<Link href="/" className={styles.logo}>
				<Image
					src="https://trakt.tv/assets/logos/header@2x-d6926a2c93734bee72c5813819668ad494dbbda651457cd17d15d267bc75c657.png"
					width={50}
					height={50}
				/>
				<div className={styles.bottomBg} />
			</Link>

			<div className={styles.menus}>
				<FiMenu color="white" size={25} strokeWidth={3} />
				<Image src="https://github.com/miguelriosoliveira.png" width={35} height={35} />
			</div>
		</header>
	);
}
