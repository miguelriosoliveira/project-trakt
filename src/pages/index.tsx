import type { NextPage } from 'next';
import Head from 'next/head';
import styles from '../styles/Home.module.css';

const Home: NextPage = () => {
  return (
    <div className={styles.container}>
      <Head>
        <title>Project Trakt</title>
        <meta
          name="description"
          content="Trakt clone made with Next.js + TypeScript and the TMDB API"
        />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main>
        <h1>hello world</h1>
      </main>
    </div>
  );
};

export default Home;
