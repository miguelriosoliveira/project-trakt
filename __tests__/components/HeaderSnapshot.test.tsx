import { render } from '@testing-library/react';

import { Header } from '../../src/components';

it('renders Header unchanged', () => {
	const { container } = render(<Header />);
	expect(container).toMatchSnapshot();
});
