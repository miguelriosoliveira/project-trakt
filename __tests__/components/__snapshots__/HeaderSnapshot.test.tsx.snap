// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header unchanged 1`] = `
<div>
  <header
    class="container"
  >
    <svg
      color="white"
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="3"
      style="color: white;"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11"
        cy="11"
        r="8"
      />
      <line
        x1="21"
        x2="16.65"
        y1="21"
        y2="16.65"
      />
    </svg>
    <a
      class="logo"
      href="/"
    >
      <img
        alt="Trakt logo: a red circle with an inclined letter T"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Ftrakt.tv%2Fassets%2Flogos%2Fheader%402x-d6926a2c93734bee72c5813819668ad494dbbda651457cd17d15d267bc75c657.png&w=96&q=75"
        srcset="/_next/image?url=https%3A%2F%2Ftrakt.tv%2Fassets%2Flogos%2Fheader%402x-d6926a2c93734bee72c5813819668ad494dbbda651457cd17d15d267bc75c657.png&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Ftrakt.tv%2Fassets%2Flogos%2Fheader%402x-d6926a2c93734bee72c5813819668ad494dbbda651457cd17d15d267bc75c657.png&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </a>
    <div
      class="menus"
    >
      <svg
        color="white"
        fill="none"
        height="25"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="3"
        style="color: white;"
        viewBox="0 0 24 24"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="3"
          x2="21"
          y1="12"
          y2="12"
        />
        <line
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <line
          x1="3"
          x2="21"
          y1="18"
          y2="18"
        />
      </svg>
      <img
        alt="The user's profile picture"
        data-nimg="1"
        decoding="async"
        height="35"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fgithub.com%2Fmiguelriosoliveira.png&w=96&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Fmiguelriosoliveira.png&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Fmiguelriosoliveira.png&w=96&q=75 2x"
        style="color: transparent;"
        width="35"
      />
    </div>
  </header>
</div>
`;
